### PR TITLE
Add types and updated dependencies

### PR DESCRIPTION
**Pull request by rubenhensen**
I've never made a NPM package before and I'm new with working with Typescript. But I believe these changes add types to your NPM package so people can use Typescript with this package. Plain JS users should be able to use the package too, like they have been in the past.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -9,19 +9,19 @@ export function writable(key, initialValue) {
...
   }
 
   return {
-    set(value) {
+    set(value: T) {
       updateStorage(key, value)
       set(value)
     },
-    update(cb) {
+    update(cb: Function) {
```
Comment (joshnuss) [inline src/index.ts:24]: Should we define the callback parameters? I'm not super experiences with TS, but I see in Svelte's source they use `Updater<T>` https://github.com/sveltejs/svelte/blob/master/src/runtime/store/index.ts#L10

**Review by joshnuss — COMMENTED** [review]
Thanks for setting this up @rubenhensen!

Left one comment

File: src/index.ts (hunk 1 above)
Comment (rubenhensen) [reply src/index.ts]: Yeah I think you're right. I don't have much experience with TS either. I changed it to Updater<T>.

**Comment by rubenhensen** [comment]
There are 2 problems I'm running into. Firstly, if I try building with rollup I get a broken index.js file. But if I compile with tsc I compile to js successfully. Secondly, I seem to have broken the jest test files and I'm not sure how to fix it.

**Comment by joshnuss** [comment]
I think we need to add `typescript` as a development dependency too. When I do that the build works for me.

For the tests, it's probably because they are all `.js`. So either we'll have to convert them to `.ts` or update the tests to `import` from the `dist`.

I think converting the tests to `.ts` makes more sense.
To do that, we would need to add `types/jest` and `jest-ts` and update the `jest` section in `package.json`:

```javascript
  "jest": {
    "testMatch": [
      "<rootDir>/test/*.test.ts"
    ],
    "transform": {
      "^.+\\.ts$": "ts-jest"
    }
  },
```

**Comment by joshnuss** [comment]
And one more thing, we'll need to update the `eslint` rules to work with `ts` too.

**Comment by rubenhensen** [comment]
I think this might be it. It took some thinking, trying to wrap my head around TS, ESLint and Jest but I believe everything works now. I removed a lot of bloat that didn't seem necessary like the entire Rollup config.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,43 +1,18 @@
...
-    ],
-    quotes: [
-      'error',
-      'single'
-    ],
-    semi: [
-      'error',
-      'never'
-    ]
-  }
```
Comment (joshnuss) [inline .eslintrc.js:42]: Can we keep the `rules` section? The rest of the file looks good

**Review by joshnuss — COMMENTED** [review]
Looks excellent! 

Left 2 comments, but otherwise good to go

File: test/localStorageStore.test.ts [hunk 3]
```diff
@@ -67,8 +68,8 @@ describe('writable()', () => {
   describe('subscribe()', () => {
     it('publishes updates', () => {
       const store = writable('myKey', 123)
-      const values = []
-      const unsub = store.subscribe(value => values.push(value))
+      const values: number[] = [] 
+      const unsub = store.subscribe(value => value!== undefined ? values.push(value) : values)
```
[the remark this answers is not in the document]
Comment (rubenhensen) [reply test/localStorageStore.test.ts]: Done

File: .eslintrc.js (hunk 2 above)
Comment (rubenhensen) [reply .eslintrc.js]: Done

**Comment by joshnuss** [comment]
Thanks @rubenhensen!!
Awesome work, and much appreciated.

I merged it manually with command line.

**Comment by joshnuss** [comment]
Published in version 0.1.5

**Pull request closed without merging**